src/explorer/WebAppTreeItem.ts

```diff
@@ -48,7 +51,15 @@ export class WebAppTreeItem extends SiteTreeItem {
         const tier: string = String(appServicePlan.sku!.tier);
         // tslint:disable-next-line:no-non-null-assertion
         this.deploymentSlotsNode = /^(basic|free|shared)$/i.test(tier) ? new DeploymentSlotsNATreeItem(tier, appServicePlan.id!) : new DeploymentSlotsTreeItem(this.client);
-        return [this.deploymentSlotsNode, this.folderNode, this.logFolderNode, this.webJobsNode, this.appSettingsNode];
+        const nodes = [this.deploymentSlotsNode, this.folderNode, this.logFolderNode, this.webJobsNode, this.appSettingsNode];
+        if (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length === 1) {
+            const workspaceConfig = vscode.workspace.getConfiguration(constants.extensionPrefix);
+            if (workspaceConfig.get(constants.configurationSettings.enableConnectionsNode)) {
+                this.connectionsNode = new ConnectionsTreeItem(this.client);
```

Comment:
You only need to create this node once in the constructor (like the other nodes are doing), rather than creating it every time "loadMoreChildren" is called